src/drive/web/modules/views/Upload/UploaderComponent.tsx

```diff
@@ -0,0 +1,236 @@
+import { connect, useDispatch } from 'react-redux'
+import { useNavigate } from 'react-router-dom'
+import { Query, useClient } from 'cozy-client'
+import Alerter from 'cozy-ui/transpiled/react/deprecated/Alerter'
+import { FixedDialog } from 'cozy-ui/transpiled/react/CozyDialogs'
+import { useI18n } from 'cozy-ui/transpiled/react'
+import {
+  purgeUploadQueue,
+  RECEIVE_UPLOAD_ERROR,
+  removeFileToUploadQueue,
+  uploadFilesFromNative
+} from 'drive/web/modules/upload'
+import { ROOT_DIR_ID } from 'drive/constants/config'
+import Header from 'drive/web/modules/move/Header'
+import Explorer from 'drive/web/modules/move/Explorer'
+import FileList from 'drive/web/modules/move/FileList'
+import Loader from 'drive/web/modules/move/Loader'
+import LoadMore from 'drive/web/modules/move/LoadMore'
+import Footer from 'drive/web/modules/move/Footer'
+import Topbar from 'drive/web/modules/move/Topbar'
+import { buildMoveOrImportQuery, buildOnlyFolderQuery } from '../../queries'
+import { generateForQueue } from './UploadUtils'
+import { DumbUploadProps, FileFromNative, Folder } from './UploadTypes'
+import { ThunkDispatch } from 'redux-thunk'
+import { useUploadFromFlagship } from './useUploadFromFlagship'
+import { useWebviewIntent } from 'cozy-intent'
+
+const TypedUseI18n = useI18n as () => {
+  t: (str: string, arg?: Record<string, unknown>) => string
+}
+const TypedAlerter = Alerter as { success: (str: string) => void }
+const TypedLoadMore = LoadMore as React.FC<{
+  hasMore: boolean
+  fetchMore: () => void
+}>
+
+const DumbUpload = ({
+  onCancel,
+  uploadFilesFromNative,
+  removeFileToUploadQueue
+}: DumbUploadProps): JSX.Element | null => {
+  const [folder, setFolder] = useState<Folder>({ _id: ROOT_DIR_ID })
+  const [uploadInProgress] = useState<boolean>(false)
+  const { t } = TypedUseI18n()
+  const navigate = useNavigate()
+  const client = useClient()
+  const { items, uploadFilesFromFlagship } = useUploadFromFlagship()
+  const webviewIntent = useWebviewIntent()
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    const onFileUpload = ({
+      data: { file, isSuccess, isLast }
+    }: {
+      data: {
+        file: FileFromNative
+        isSuccess: boolean
+        isLast?: boolean
+      }
+    }): void => {
+      if (!file) return
+
+      if (!isSuccess) {
+        dispatch({ type: RECEIVE_UPLOAD_ERROR, file })
+        return
+      }
+
+      removeFileToUploadQueue({ name: file.fileName })
+        .then(() => {
+          if (isLast) {
+            return TypedAlerter.success(t('ImportToDrive.success'))
+          }
+          return
+        })
+        .catch(e => {
+          console.log('removeFileToUploadQueue error', e)
+        })
+    }
+
+    window.addEventListener('message', onFileUpload)
+  }, [dispatch, removeFileToUploadQueue, t])
+
+  const onClose = useCallback(async () => {
+    await webviewIntent?.call('resetFilesToHandle').then(() => {
```

Comment:
I still really don't like this way of writing code... Can we do it differently to be the same as other place?